nixos/modules/services/continuous-integration/gocd-agent/default.nix

```diff
@@ -0,0 +1,147 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.gocd-agent;
+in {
+  options = {
+    services.gocd-agent = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable the gocd-agent continuous integration server.
+        '';
+      };
+
+      user = mkOption {
+        default = "gocd-agent";
+        type = types.str;
+        description = ''
+          User the gocd-agent server should execute under.
+        '';
+      };
+
+      group = mkOption {
+        default = "gocd-agent";
+        type = types.str;
+        description = ''
+          If the default user "gocd-agent" is configured then this is the primary
+          group of that user.
+        '';
+      };
+
+      extraGroups = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        example = [ "wheel" "docker" ];
+        description = ''
+          List of extra groups that the "gocd-agent" user should be a part of.
+        '';
+      };
+
+      packages = mkOption {
+        default = [ pkgs.stdenv pkgs.jre7 config.programs.ssh.package pkgs.nix ];
+        type = types.listOf types.package;
+        description = ''
+          Packages to add to PATH for the gocd-agent process.
+        '';
+      };
+
+      goServer = mkOption {
+        default = "127.0.0.1";
+        type = types.str;
+        description = ''
+          Address of GoCD Server to attach the GoCD Agent to.
+        '';
+      };
+
+      goServerPort = mkOption {
+        default = 8153;
+        type = types.int;
+        description = ''
+          Port that GoCD Server is Listening on.
+        '';
+      };
+
+      workDir = mkOption {
+        default = "/var/lib/go-agent";
+        type = types.str;
+        description = ''
+          Specifies the working directory in whcih the gocd-agent java archive resides.
+          The default is /var/lib/go-agent.
+        '';
+      };
+
+      environment = mkOption {
+        default = { };
+        type = with types; attrsOf str;
+        description = ''
+          Additional environment variables to be passed to the gocd-agent process.
+          As a base environment, gocd-agent receives NIX_PATH from
+          <option>environment.sessionVariables</option>, NIX_REMOTE is set to
+          "daemon".
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.extraGroups = optional (cfg.group == "gocd-agent") {
+      name = "gocd-agent";
+    };
+
+    users.extraUsers = optional (cfg.user == "gocd-agent") {
```

Comment:
Please provide UID and GID here https://github.com/swarren83/nixpkgs/blob/f4a5d0829afef15089ce66d5016137df127c715c/nixos/modules/misc/ids.nix